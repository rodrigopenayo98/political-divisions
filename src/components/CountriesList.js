import { useSelector } from 'react-redux';
import { Country } from './Country';
import StatusType from './StatusTypes';

function CountriesList() {
  const countries = useSelector((store) => store.countries);

  if (!countries || countries.status === StatusType.LOADING) {
    return <span>Cargando...</span>;
  }

  return (
    <section>
      <ul>
        {countries?.countriesArr.map((country, index) => (
          <Country
            key={country.geonameId}
            geonameId={country.geonameId}
            index={index}
            countryName={country.countryName}
            countryCode={country.countryCode}
            currencyCode={country.currencyCode}
            areaInSqKm={country.areaInSqKm}
            continentName={country.continentName}
            population={country.population}
          />
        ))}
      </ul>
    </section>
  );
}

export default CountriesList;
