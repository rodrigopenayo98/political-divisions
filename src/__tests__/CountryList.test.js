import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-dom'; // Importamos MemoryRouter
import CountriesList from '../components/CountriesList';
import StatusType from '../components/StatusTypes';

const mockStore = configureStore([]);

test('CountriesList component snapshot', () => {
  const initialState = {
    countries: {
      countriesArr: [
        {
          geonameId: 1,
          countryName: 'United States',
          countryCode: 'US',
          currencyCode: 'USD',
          areaInSqKm: '9372610',
          continentName: 'North America',
          population: '331002651',
        },
      ],
      searchString: 'United',
      status: StatusType.SUCCESS,
    },
  };
  const store = mockStore(initialState);

  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <CountriesList />
        </MemoryRouter>
      </Provider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
