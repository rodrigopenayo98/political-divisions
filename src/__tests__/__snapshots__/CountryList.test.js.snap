// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountriesList component snapshot 1`] = `
<section>
  <ul>
    <li>
      <div
        className="country-card bg-green-4"
      >
        <a
          className="country-info"
          href="/country/1"
          onClick={[Function]}
        >
          <img
            alt="United States"
            className="country-image"
            src="https://flagcdn.com/us.svg"
          />
          <p
            className="country-name"
          >
            United States
          </p>
          <p
            className="currency-code"
          >
            USD
          </p>
        </a>
        <div
          className="country-details"
        >
          <p
            className="area"
          >
            Superficie:
             
            9372610
             
            km²
          </p>
          <p
            className="continent"
          >
            Continente:
             
            North America
          </p>
          <p
            className="population"
          >
            Población:
             
            331002651
          </p>
        </div>
      </div>
    </li>
  </ul>
</section>
`;
